### PR TITLE
polish: improve consistency of warnings and errors

### PR DESCRIPTION
**Pull request by petebacondarwin**
Related to #377

Screenshot 2022-04-26 at 13 57 43

### CODE REVIEW

**Comment by threepointone** [comment]
Ooh that tower of icons looks a bit imposing, no?

File: packages/wrangler/src/logger.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { format } from "node:util";
```
Comment (JacobMGEvans) [inline packages/wrangler/src/logger.ts:1]: Could this use the ESbuild formatter?

Comment (petebacondarwin) [reply packages/wrangler/src/logger.ts:1]: This is to account for the fact that `console.log` and friends do fancy formatting based on the first string in the parameter list. E.g. `console.log("pre%spost", "foo")` will output `prefoopost`.

But we could then take the output of `format` and pass it through esbuild's `formatMessagesSync()` possibly.

**Comment by petebacondarwin** [comment]
> Ooh that tower of icons looks a bit imposing, no?

I actually like how it clearly delineates the warnings and errors from the informative messages, which have no icon.
In most cases there should be no errors, so you would not see any Xs.
In many cases there should be no warnings. And if there are it is nice to know where the warning ends and the other messages start.

**Comment by petebacondarwin** [comment]
The layout suggested by @JacobMGEvans could only be generated if we captured the calls to `logger` and only rendered them at some point (?) once they had all been completed.
We already do this, effectively, with validation warnings and errors because they are captured in `Diagnostics` messages, which are then all rendered together after validation completes.

**Comment by petebacondarwin** [comment]
I updated to use the esbuild formatter. I am not sure it is any better but open to opinions...

Screenshot 2022-04-26 at 20 26 30

**Comment by threepointone** [comment]
I like this better!

**Comment by petebacondarwin** [comment]
OK let's go with this. I just need to work out how to fix the Windows snapshots 😿

File: packages/wrangler/src/logger.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/logger.ts:1]: > But we could then take the output of format and pass it through esbuild's formatMessagesSync() possibly.

I like this idea, also just saw this `formatMessagesSync()` in the TS interface today and thought you would like the sync version 😆

**Review by threepointone — APPROVED** [review]

**Comment by petebacondarwin** [comment]
I will follow up with a PR to change the mocking to mock the logger rather than the console

**Pull request merged**